src/berry_mill/mill.py

```diff
@@ -81,6 +81,10 @@ def run(self) -> None:
         Build an image
         """
 
+        # if target_dir exists with no --clean option, fail early.
+        if not self.args.clean and os.path.isdir(self.args.target_dir):
+            raise Exception("Target directory already exists. Missed --clean ?")
```

Comment:
Maybe this would be a bit better:
`"Target directory already exists. Hint: use --clean option."`